src/ui/public/kbn_top_nav/kbn_top_nav_controller.js

```diff
@@ -89,9 +90,12 @@ export function KbnTopNavControllerProvider($compile) {
     // apply the defaults to individual options
     _applyOptDefault(opt = {}) {
       const defaultedOpt = {
-        label: capitalize(opt.key),
+        label: opt.label ? opt.label : capitalize(opt.key),
         hasFunction: !!opt.run,
-        description: opt.run ? opt.key : `Toggle ${opt.key} view`,
+        description: opt.run ? opt.key : i18n.translate('common.ui.topNav.toggleViewAriaLabel', {
+          defaultMessage: 'Toggle {optKey} view',
+          values: { optKey: opt.key }
```

Comment:
We should use the `label` here if present and not the `key` if possible, since the key will now never be translated.